engine/render/src/render/program_utils.cpp

```diff
@@ -236,23 +248,38 @@ namespace dmRender
             if (uniform_desc.m_Type == dmGraphics::TYPE_FLOAT_VEC4 || uniform_desc.m_Type == dmGraphics::TYPE_FLOAT_MAT4)
             {
                 HConstant render_constant = dmRender::NewConstant(uniform_desc.m_NameHash);
+
                 dmRender::SetConstantLocation(render_constant, uniform_desc.m_Location);
 
-                if (uniform_desc.m_Type == dmGraphics::TYPE_FLOAT_MAT4)
+                // We are be about to add a duplicate. Make sure to reuse the data
+                HConstant prev_constant = FindConstant(constants, uniform_desc.m_NameHash);
+                if (prev_constant != 0)
```

Comment:
When creating the initial material constants (when loading the material), we check if there are duplicates.
If so, we make sure it uses the same memory as the first constant (with the same name).

This makes sure we can have multiple uniforms (vs + fs) with different locations, but still use get/set the same values.